### PR TITLE
[cmake] add support for nrf528xx

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
It's based on https://github.com/openthread/openthread/pull/4767

### CODE REVIEW

File: examples/platforms/nrf528xx/nrf52811/arm-none-eabi.cmake [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+set(CMAKE_CXX_COMPILER             arm-none-eabi-g++)
+set(CMAKE_ASM_COMPILER             arm-none-eabi-as)
+set(CMAKE_RANLIB                   arm-none-eabi-ranlib)
+
+execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE COMPILER_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+set(COMMON_C_FLAGS                 "-mcpu=cortex-m4 -mfloat-abi=soft -mthumb -mabi=aapcs -fdata-sections -ffunction-sections")
+
+if(COMPILER_VERSION VERSION_GREATER_EQUAL 7)
+    set(COMMON_C_FLAGS                 "${COMMON_C_FLAGS} -Wno-expansion-to-defined")
```
Comment (LuDuda) [inline examples/platforms/nrf528xx/nrf52811/arm-none-eabi.cmake:42]: nit: alignment

**Review by LuDuda — APPROVED** [review]
This looks neat! LGTM 👍

File: examples/platforms/nrf528xx/nrf52833/arm-none-eabi.cmake [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (Irving-cl) [inline examples/platforms/nrf528xx/nrf52833/arm-none-eabi.cmake:2]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: examples/platforms/nrf528xx/nrf52833/nrf52833.cmake [hunk 3]
```diff
@@ -0,0 +1,220 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (Irving-cl) [inline examples/platforms/nrf528xx/nrf52833/nrf52833.cmake:2]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: script/cmake-build [hunk 4]
```diff
@@ -159,6 +198,55 @@ main()
...
+                soft_crypto)
+                    options+=("${local_options[@]}")
+                    ;;
+                soft_crypto_threading)
+                    options+=("${local_options[@]}" "-DOT_MBEDTLS_THREADING=ON")
+                    ;;
+            esac
+            ;;
         *)
             options+=("-DCMAKE_TOOLCHAIN_FILE=examples/platforms/${platform}/arm-none-eabi.cmake")
```
Comment (Irving-cl) [inline script/cmake-build:251]: Remove this line? The TOOLCHAIN_FILE has been specified above for nrf (`"-DCMAKE_TOOLCHAIN_FILE=examples/platforms/nrf528xx/${platform}/arm-none-eabi.cmake"`).

File: third_party/jlink/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,61 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (Irving-cl) [inline third_party/jlink/CMakeLists.txt:2]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: third_party/NordicSemiconductor/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,375 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (Irving-cl) [inline third_party/NordicSemiconductor/CMakeLists.txt:2]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: examples/platforms/nrf528xx/nrf52840/nrf52840.cmake [hunk 7]
```diff
@@ -0,0 +1,233 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (Irving-cl) [inline examples/platforms/nrf528xx/nrf52840/nrf52840.cmake:2]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: examples/platforms/nrf528xx/nrf52840/arm-none-eabi.cmake [hunk 8]
```diff
@@ -0,0 +1,56 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (Irving-cl) [inline examples/platforms/nrf528xx/nrf52840/arm-none-eabi.cmake:2]:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: script/cmake-build (hunk 4 above)
Comment (lmaciejonczyk) [reply script/cmake-build:251]: It's not part of the nrf build. It is for other platform which are not mentioned here clearly.

Comment (Irving-cl) [reply script/cmake-build:251]: Thanks. Sorry for the mistake.

**Review by Irving-cl — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]
Thanks @lmaciejonczyk !

**Pull request merged**